bindings/c/Cargo.toml

```diff
@@ -28,7 +28,7 @@ rust-version.workspace = true
 version.workspace = true
 
 [lib]
-crate-type = ["cdylib"]
+crate-type = ["cdylib", "staticlib"]
```

Comment:
It's better to link with libopendal_c with `cdylib`.

Reply:
`cdylib` has some pitfalls on Apple platforms, such as `swift test` may not find the dylib file without tweaking the rpath. But currently, `staticlib` will also cause troubles for other bindings (eg. zig needs additional linking flags to link libunwind correctly). Maybe we can create a `staticlib` target only for Swift binding, under `swift` directory. What do you think?

Reply:
> Maybe we can create a `staticlib` target only for Swift binding, under `swift` directory. What do you think?

Not a good idea to me. I prefer to fix from zig side. Tracked at https://github.com/apache/incubator-opendal/issues/2471

Reply:
Fixed in https://github.com/apache/incubator-opendal/pull/2473

Reply:
Please merge with main to get the fix.